main.tf

```
@@ -1 +1,445 @@
+
+  name           = var.name
+  read_capacity  = 5
+  write_capacity = 5
+  hash_key       = "Path"
+  range_key      = "Key"
+
+  attribute {
+    name = "Path"
+    type = "S"
+  }
+
+  attribute {
+    name = "Key"
+    type = "S"
+  }
+
+  tags = merge({ Name = var.name }, local.tags)
+}
+
+resource "aws_appautoscaling_target" "this" {
+  count = var.dynamodb_table == null ? 1 : 0
+
+  max_capacity       = var.dynamodb_max_read_capacity
+  min_capacity       = var.dynamodb_min_read_capacity
+  resource_id        = join("/", ["table", local.dynamodb_table])
+  scalable_dimension = "dynamodb:table:ReadCapacityUnits"
+  service_namespace  = "dynamodb"
+}
+
+resource "aws_appautoscaling_policy" "this" {
+  count = var.dynamodb_table == null ? 1 : 0
+
+  name               = join(":", ["DynamoDBReadCapacityUtilization", join("", aws_appautoscaling_target.this.*.resource_id)])
+  policy_type        = "TargetTrackingScaling"
+  resource_id        = join("", aws_appautoscaling_target.this.*.resource_id)
+  scalable_dimension = join("", aws_appautoscaling_target.this.*.scalable_dimension)
+  service_namespace  = join("", aws_appautoscaling_target.this.*.service_namespace)
+
+  target_tracking_scaling_policy_configuration {
+    predefined_metric_specification {
+      predefined_metric_type = "DynamoDBReadCapacityUtilization"
+    }
+
+    target_value = var.dynamodb_target_value
+  }
+}
+
+# Manage autoscaling group
+module "autoscaling_group" {
+  source = "git::https://github.com/plus3it/terraform-aws-watchmaker//modules/lx-autoscale?ref=1.15.7"
+
+  Name            = var.name
+  OnFailureAction = ""
+  DisableRollback = "true"
+
+  AmiId                = data.aws_ami.this.id
+  AmiDistro            = "CentOS"
+  AppScriptUrl         = join("", ["s3://", local.appscript_url])
+  CfnBootstrapUtilsUrl = var.cfn_bootstrap_utils_url
+
+  CfnEndpointUrl     = var.cfn_endpoint_url
+  CloudWatchAgentUrl = var.cloudwatch_agent_url
+  CloudWatchAppLogs  = local.logs
+  KeyPairName        = var.key_pair_name
+  InstanceRole       = module.iam.profile_name
+  InstanceType       = var.instance_type
+  NoReboot           = true
+  NoPublicIp         = true
+  PypiIndexUrl       = var.pypi_index_url
+  SecurityGroupIds   = join(",", compact(concat([aws_security_group.ec2.id], var.ec2_extra_security_group_ids)))
+  SubnetIds          = join(",", var.ec2_subnet_ids)
+  TargetGroupArns    = aws_lb_target_group.this.arn
+  ToggleNewInstances = var.toggle_update
+  TimeoutInMinutes   = "20"
+
+  WatchmakerEnvironment = var.environment
+  WatchmakerConfig      = var.watchmaker_config
+  WatchmakerAdminGroups = var.watchmaker_admin_groups
+  WatchmakerAdminUsers  = var.watchmaker_admin_users
+  WatchmakerOuPath      = var.watchmaker_ou_path
+
+  DesiredCapacity = var.desired_capacity
+  MinCapacity     = var.min_capacity
+  MaxCapacity     = var.max_capacity
+
+  EnableRepos = join(" ", concat(var.enabled_repos, local.default_enabled_repos))
```

`default` implies that the user can override the value, but this implementation does not allow that.

i'd suggest changing the definition of `local.default_enabled_repos`:

```
locals {
  enable_repos = var.enable_repos == null ? "epel" : var.enable_repos
```

```suggestion
  EnableRepos = local.enable_repos
```